### PR TITLE
Update README for typos, better Error examples, and whitespace.

### PR DESCRIPTION
**Pull request by jshier**
This PR updates the README for typos, better Error examples, and whitespace.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -333,9 +333,9 @@ Alamofire.request("https://httpbin.org/get")
     .validate(contentType: ["application/json"])
     .response { response in
 	    switch response.result {
-	    case .Success:
+	    case .success:
     	    print("Validation Successful")
-	    case .Failure(let error):
+	    case .failure(let error):
```
Comment (cnoon) [inline README.md]: Nice catch!

File: README.md [hunk 2]
```diff
@@ -348,9 +348,9 @@ Automatically validates status code within `200...299` range, and that the `Cont
 ```swift
 Alamofire.request("https://httpbin.org/get").validate().responseJSON { response in
     switch response.result {
-    case .Success:
+    case .success:
         print("Validation Successful")
-    case .Failure(let error):
+    case .failure(let error):
```
Comment (cnoon) [inline README.md]: And again!

File: README.md [hunk 3]
```diff
@@ -983,7 +983,7 @@ Generally speaking, either the default implementation or the override closures s
 
 ### Request
 
-The result of a `request`, `download`, `upload` or `stream` methods are a `DataRequest`, `DownloadRequest`, `UploadRequest` and StreamRequest` which all inherit from `Request`. All `Request` instances are always created by an owning session manager, and never initialized directly.
+The result of a `request`, `download`, `upload` or `stream` methods are a `DataRequest`, `DownloadRequest`, `UploadRequest` and `StreamRequest` which all inherit from `Request`. All `Request` instances are always created by an owning session manager, and never initialized directly.
```
Comment (cnoon) [inline README.md]: Great catch!

File: README.md [hunk 4]
```diff
@@ -1009,7 +1009,7 @@ let url = URL(string: urlString)!
 Alamofire.request(url, method: .post)
 
 let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)
-Alamofire.request(.POST, URLComponents)
+Alamofire.request(.post, URLComponents)
```
Comment (cnoon) [inline README.md]: Whoops!

File: README.md [hunk 5]
```diff
@@ -1500,19 +1502,17 @@ extension DataRequest {
             guard error == nil else { return .failure(BackendError.network(error: error!)) }
 
             let jsonSerializer = DataRequest.jsonResponseSerializer(options: .allowFragments)
-            let result = jsonSerializer.serializeResponse(request, response, data, error)
+            let result = jsonSerializer.serializeResponse(request, response, data, nil)
```
Comment (cnoon) [inline README.md]: Newline?

File: README.md [hunk 6]
```diff
@@ -1365,10 +1365,14 @@ For example, here's how a response handler using [Ono](https://github.com/mattt/
...
     static func xmlResponseSerializer() -> DataResponseSerializer<ONOXMLDocument> {
         return DataResponseSerializer { request, response, data, error in
+            // Pass through any underlying URLSession error to the .network case.
             guard error == nil else { return .failure(BackendError.network(error: error!)) }
 
-            guard let validData = data else {
-                return .failure(BackendError.dataSerialization(reason: "Input data was nil."))
+            // Use Alamofire's existing data serializer to extract the data, passing the error as nil, as it has
+            // alreaady been handled.
+            let result = Request.serializeResponseData(response: response, data: data, error: nil)
```
Comment (cnoon) [inline README.md]: Newline?

File: README.md [hunk 7]
```diff
@@ -1414,18 +1418,16 @@ extension DataRequest {
             guard error == nil else { return .failure(BackendError.network(error: error!)) }
 
             let jsonResponseSerializer = DataRequest.jsonResponseSerializer(options: .allowFragments)
-            let result = jsonResponseSerializer.serializeResponse(request, response, data, error)
+            let result = jsonResponseSerializer.serializeResponse(request, response, data, nil)
```
Comment (cnoon) [inline README.md]: Newline?

File: README.md (hunk 5 above)
Comment (jshier) [inline README.md]: Added in [0a05e2b](https://github.com/Alamofire/Alamofire/commit/880dc772a12d1ec490eb8aee029115ec8).

File: README.md (hunk 7 above)
Comment (jshier) [inline README.md]: Added in [0a05e2b](https://github.com/Alamofire/Alamofire/commit/880dc772a12d1ec490eb8aee029115ec8).

File: README.md (hunk 6 above)
Comment (jshier) [inline README.md]: Added in [0a05e2b](https://github.com/Alamofire/Alamofire/commit/880dc772a12d1ec490eb8aee029115ec8).

**Comment by cnoon** [comment]
Looks great @jshier...I'm gonna merge it.

**Pull request merged**